### ISSUE TITLE
puppetParser does not descend into classes to generate tags for variable definitions. 

### ISSUE THREAD

**Issue by ahakanbaba** [universal-ctags/ctags#1905]
*****

The name of the parser: puppetManifest 

The command line you used to run ctags:

```
echo input.pp  | ctags --options=NONE -L -
```

The content of input file:

```
$_msg = 'hello, world!'
class helloworld {
    notify { $::_msg: }

  $msg_in_class = 'hello, world!'
}
```

The tags output you are not satisfied with:

```
_msg	input.pp	/^$_msg = 'hello, world!'$/;"	v
helloworld	input.pp	/^class helloworld {$/;"	c
```

The tags output you expect:

```
_msg	input.pp	/^$_msg = 'hello, world!'$/;"	v
helloworld	input.pp	/^class helloworld {$/;"	c
msg_in_class	input.pp  /^    $msg_in_class = 'hello, world!'$/;"  	v
```
(The 3rd line is the tag for the variable declaration inside a class. I may have a mistake in the regex there ) 

The version of ctags:

```
$ ctags --version
Universal Ctags 0.0.0(b6b46df1), Copyright (C) 2015 Universal Ctags Team
Universal Ctags is derived from Exuberant Ctags.
Exuberant Ctags 5.8, Copyright (C) 1996-2009 Darren Hiebert
  Compiled: Oct  4 2018, 12:52:40
  URL: https://ctags.io/
  Optional compiled features: +wildcards, +regex, +iconv, +debug, +option-directory, +xpath, +yaml, +case-insensitive-filenames
```

How do you get ctags binary:

Building it locally, in macOs ElCapitan

**Comment by ahakanbaba** [comment]
The input file is indeed valid puppet code 

```
$ cat dummy.pp
$_msg = 'hello, world!'
class helloworld {
      notify { $::_msg: }

        $msg_in_class = 'hello, world!'
}

$ puppet  apply --noop dummy.pp
Notice: Compiled catalog for hbaba.dev.box.net in environment production in 0.07 seconds
Notice: Applied catalog in 0.07 seconds
```

**Comment by masatake** [comment]
Try #1910.

**Comment by ahakanbaba** [comment]
With your changes in #1910 I was able to get the expected output. Thanks 

```
$ cat example_1905.pp
$_msg = 'hello, world!'
class helloworld {
    notify { $::_msg: }

  $msg_in_class = 'hello, world!'
}
$ ctags -o - example_1905.pp
_msg	example_1905.pp	/^$_msg = 'hello, world!'$/;"	v
helloworld	example_1905.pp	/^class helloworld {$/;"	c
msg_in_class	example_1905.pp	/^  $msg_in_class = 'hello, world!'$/;"	v	class:helloworld
```

**Comment by masatake** [comment]
Fixed with #1910.
